File: app/Models/Lead.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Lead extends Model
{
    protected $fillable = ['customer_id', 'assigned_employee_id', 'source', 'lead_status', 'notes'];
    public function customer()
    {
        return $this->belongsTo(Customer::class);
    }
    public function assignee()
    {
        return $this->belongsTo(Employee::class, 'assigned_employee_id');
    }
}

File: lib/models/lead.dart
class Lead {
  final int id;
  final int? customerId;
  final int? assignedEmployeeId; 
  final String? source;
  final String? leadStatus;
  final DateTime dateCreated;

  Lead({
    required this.id,
    this.customerId,
    this.assignedEmployeeId,
    this.source,
    this.leadStatus,
    required this.dateCreated,
  });

  factory Lead.fromRow(Map<String, dynamic> r) => Lead(
    id: r['id'] as int,
    customerId: r['customer_id'] as int?,
    assignedEmployeeId: r['assigned_employee_id'] as int?,
    source: r['source'] as String?,
    leadStatus: r['lead_status'] as String?,
    dateCreated: DateTime.parse(r['date_created'] as String),
  );
}
